### PR TITLE
[IMP] mail, im_livechat: remove conditions from sort

### PR DESCRIPTION
**Pull request by oomsveta**
This commit makes all sorts unconditional, which would allow us to get rid of the functions that wrap the sorting rules and use arrays directly instead.

This will allow us to lighten the syntax and to perform some optimizations in subsequent PRs.

Task-2992581.

### CODE REVIEW

File: addons/im_livechat/static/src/models/discuss_sidebar_category.js [hunk 1]
```diff
@@ -13,6 +13,25 @@ addFields('DiscussSidebarCategory', {
     }),
 });
 
+patchFields('DiscussSidebarCategory', {
+    categoryItemsOrdredByLastAction: {
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/models/discuss_sidebar_category.js:17]: typo `Ordred` => `Ordered`

**Review by alexkuhn — APPROVED** [review]
LGTM otherwise

File: addons/im_livechat/static/src/models/discuss_sidebar_category.js (hunk 1 above)
Comment (oomsveta) [reply addons/im_livechat/static/src/models/discuss_sidebar_category.js]: good catch; changes done 👌

**Comment by seb-odoo** [comment]
need to close and target v16

**Comment by oomsveta** [comment]
> need to close and target v16

why would I need to close it?

**Comment by seb-odoo** [comment]
The prefix on the branch is wrong and sometimes that causes issues.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**